types/backbone/index.d.ts

```diff
@@ -129,35 +134,23 @@ declare namespace Backbone {
         * For assigning an object hash, do it like this: this.defaults = <any>{ attribute: value, ... };
         * That works only if you set it in the constructor or the initialize method.
         **/
-        defaults(): ObjectHash;
+        defaults(): Partial<TAttributes>;
         id: any;
-        idAttribute: string;
+        idAttribute: keyof TAttributes;
         validationError: any;
         urlRoot: any;
 
-        constructor(attributes?: any, options?: any);
-        initialize(attributes?: any, options?: any): void;
+        constructor(attributes?: Partial<TAttributes>, options?: any);
+        initialize(attributes?: Partial<TAttributes>, options?: any): void;
 
         fetch(options?: ModelFetchOptions): JQueryXHR;
+        parse(response: any, options?: any): Partial<TAttributes>;
 
-        /**
-        * For strongly-typed access to attributes, use the `get` method only privately in public getter properties.
-        * @example
-        * get name(): string {
-        *    return super.get("name");
-        * }
-        **/
-        /*private*/ get(attributeName: string): any;
+        get<K extends keyof TAttributes>(attributeName: K): TAttributes[K];
 
-        /**
-        * For strongly-typed assignment of attributes, use the `set` method only privately in public setter properties.
-        * @example
-        * set name(value: string) {
-        *    super.set("name", value);
-        * }
-        **/
-        /*private*/ set(attributeName: string, value: any, options?: ModelSetOptions): Model;
-        set(obj: any, options?: ModelSetOptions): Model;
+        // TODO: this still doesn't type check!!!
```

Comment:
Could you update this comment to be more explanatory?

Reply:
Sorry, I don't know why the typing on that method is broken. It *should* work...

Reply:
I might be able to help - can you give an example of how it's broken?

Reply:
When I tried it, `set` behaved like it was completely untyped. Perhaps it was a bug in the version of Typescript at the time. I haven't used this code for a couple of months.